table/merging_iterator.cc

```diff
@@ -74,13 +69,8 @@ class MergingIterator : public InternalIterator {
       iter->SetPinnedItersMgr(pinned_iters_mgr_);
     }
     auto new_wrapper = children_.back();
-    if (new_wrapper.Valid()) {
-      assert(new_wrapper.status().ok());
-      minHeap_.push(&new_wrapper);
-      current_ = CurrentForward();
-    } else {
-      considerStatus(new_wrapper.status());
-    }
+    AddToMinHeapOrCheckStatus(&new_wrapper);
+    current_ = CurrentForward();
```

Comment:
The logic here differs a bit from before in that if ```new_wrapper.Valid()``` is false, we'll still call ```CurrentForward()```.

Reply:
Good catch. Let me fix it.